### PR TITLE
Remove nssize and smallfiles startup options

### PR DESCRIPTION
**Pull request by ajdavis**
Removed in 4.2, see SERVER-35715 and SERVER-35984.

### CODE REVIEW

**Comment by xdg** [comment]
I wonder if we need to fork our configs so we can keep testing old servers
with optimised settings.

**Comment by ajdavis** [comment]
I've also removed prealloc now.

To my surprise it seems that removing nssize, smallfiles, and noprealloc has no effect on 3.0 replica set startup times. I tested on Linux and Windows in Evergreen and it takes about 24 seconds to start the RS regardless. MongoDB 2.6 is EOL and MongoDB 3.2 defaults to WiredTiger.

File: mongo_orchestration/configurations/replica_sets/allengines.json [hunk 1]
```diff
@@ -4,10 +4,7 @@
             "procParams": {
                 "ipv6": true,
                 "journal": true,
-                "storageEngine": "mmapv1",
-                "noprealloc": true,
```
Comment (behackett) [inline mongo_orchestration/configurations/replica_sets/allengines.json]: I think you want to leave these for mmapv1?

File: mongo_orchestration/configurations/replica_sets/mmapv1.json [hunk 2]
```diff
@@ -6,8 +6,7 @@
                 "journal": true,
                 "storageEngine": "mmapv1",
                 "noprealloc": true,
-                "nssize": 1,
```
Comment (behackett) [inline mongo_orchestration/configurations/replica_sets/mmapv1.json]: Leave these for mmapv1?

File: mongo_orchestration/configurations/replica_sets/mmapv1.json [hunk 3]
```diff
@@ -23,8 +22,7 @@
...
```
Comment (behackett) [inline mongo_orchestration/configurations/replica_sets/mmapv1.json]: Same here.

File: mongo_orchestration/configurations/replica_sets/mmapv1.json [hunk 4]
```diff
@@ -40,8 +38,7 @@
...
```
Comment (behackett) [inline mongo_orchestration/configurations/replica_sets/mmapv1.json]: And here.

**Comment by ajdavis** [comment]
Good catch @behackett, I didn't intend to change those MMAPv1 configurations. New patch.

**Pull request merged**